### PR TITLE
feat: create clone modal

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [ ] Not that difficult
- [ ] Approved feature branch merge to master

### Description
스크린샷 2024-09-19 오후 5 27 40

### CODE REVIEW

**Review by sulmoJ — APPROVED** [review]
LGTM~

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+/* Util */
+const getUniqueFolderName = (name: string): string => {
+    const _newName = `Clone - ${name}`;
+    return dashboardMainPageGetters.existingFolderNameList.find((n) => n === _newName) ? getUniqueFolderName(_newName) : _newName;
+};
+const getUniqueDashboardName = (name: string): string => {
+    const _newName = `Clone - ${name}`;
+    return dashboardMainPageGetters.existingDashboardNameList.find((n) => n === _newName) ? getUniqueDashboardName(_newName) : _newName;
+};
+
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue:73]: idea: These utils would be useful if it is in util package :)

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        const createdFolder = await fetcher(params);
+        dashboardMainPageStore.setNewIdList([
+            ...dashboardMainPageState.newIdList,
+            createdFolder.folder_id as string,
+        ]);
+        return createdFolder.folder_id;
+    } catch (e) {
+        return undefined;
+    }
+};
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue:108]: question: what happens if creation fails? 😮

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                ..._sharedDashboard,
+                resource_group: RESOURCE_GROUP.DOMAIN,
+            };
+        } else if (!isPrivate) {
+            _sharedDashboard = {
+                ..._sharedDashboard,
+                resource_group: dashboard?.resource_group || RESOURCE_GROUP.WORKSPACE,
+            };
+        }
+        const createdDashboard = await dashboardStore.createDashboard(_dashboardType, _sharedDashboard);
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue:133]: nitpick: I think.. making new object everytime is pretty inefficient.

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+};
+
+/* Event */
+const handleCloneConfirm = async () => {
+    state.loading = true;
+    const _createDashboardPromises: Promise<boolean>[] = [];
+
+    await Promise.all(dashboardMainPageGetters.selectedTreeData.map(async (item) => {
+        if (item.data.type === 'FOLDER') {
+            const _isPrivate = state.privateMap[item.data.id];
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue:151]: issue: Korean comments.. 🥲

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+            const createdFolderId = await createFolder(item.data.name, _isPrivate);
+            item.children.forEach((child) => {
+                _createDashboardPromises.push(createDashboard(child.data, _isPrivate, createdFolderId));
+            });
+        } else {
+            _createDashboardPromises.push(createDashboard(item.data, false));
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue:157]: issue: I think this flow can cause critical issue.
`createdFolderId` can be undefined when api request has failed.
Even though users wanted to make dashboard under specific folder, in this case, the dashboard would be created on root.

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+        }
+    }));
+    const _results = await Promise.all(_createDashboardPromises);
+
+    if (_results.every((r) => r)) {
+        showSuccessMessage(i18n.t('DASHBOARDS.ALL_DASHBOARDS.ALT_S_CLONE_DASHBOARD'), '');
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue:163]: question: If one has failed, then all the rest must be stopped?

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+    } else {
+        ErrorHandler.handleRequestError(new Error('Delete failed'), i18n.t('DASHBOARDS.ALL_DASHBOARDS.ALT_E_CLONE_DASHBOARD'));
+    }
+    await dashboardStore.load();
+    dashboardMainPageStore.setSelectedIdMap({});
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue:168]: issue: 'Delete failed' << I think it's typo

**Review by WANZARGEN — COMMENTED** [review]
Great job on this! I've left a few small suggestions that might help improve the code. Let me know what you think!

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue (hunk 1 above)
Comment (yuda110) [reply apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue]: I made it!

**Pull request merged**